kvmagent/kvmagent/plugins/zbs_storage_plugin.py

```
@@ -4,23 +4,235 @@
+    def _is_luks_volume(self, install_path, encrypted_dek=None):
+        if encrypted_dek:
+            with volume_secret.luks_secret_channel(encrypted_dek) as secret_file:
+                cmd = '/usr/bin/qemu-img info --object secret,id=luks_sec,format=raw,file=%s %s' % (
+                    linux.shellquote(secret_file), self._luks_image_opts(install_path))
+                bash.bash_errorout(cmd)
+                return True
+        out = bash.bash_errorout('/usr/bin/qemu-img info %s' % linux.shellquote(self._cbd_qemu_path(install_path)))
+        return 'file format: luks' in out
+
+    def _new_luks_temp_path(self):
+        return '/tmp/zbs-luks-%s.img' % uuid.uuid4().hex
```

Ruff (0.16.0)

[error] Probable insecure usage of temporary file or directory: "/tmp/zbs-luks-%s.img"

(S108)